test/integration/validation/certificaterequest_test.go

```diff
@@ -0,0 +1,155 @@
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"context"
+	"encoding/pem"
+	"strings"
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/jetstack/cert-manager/pkg/api"
+	v1 "github.com/jetstack/cert-manager/pkg/apis/certmanager/v1"
+	cmmeta "github.com/jetstack/cert-manager/pkg/apis/meta/v1"
+	"github.com/jetstack/cert-manager/pkg/util/pki"
+	"github.com/jetstack/cert-manager/test/integration/framework"
+)
+
+var certGVK = schema.GroupVersionKind{
+	Group:   "cert-manager.io",
+	Version: "v1",
+	Kind:    "CertificateRequest",
+}
+
+func TestValidationCertificateRequests(t *testing.T) {
+	tests := map[string]struct {
+		input       runtime.Object
+		errorSuffix string // is a suffix as the API server sends the whole value back in the error
+		expectError bool
+	}{
+		"No errors on valid certificaterequest with no usages set": {
+			input: &v1.CertificateRequest{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test",
+					Namespace: "default",
+				},
+				Spec: v1.CertificateRequestSpec{
+					Request: mustGenerateCSR(t, &v1.Certificate{
+						Spec: v1.CertificateSpec{
+							DNSNames: []string{"example.com"},
+						},
+					}),
+					Usages:    []v1.KeyUsage{},
+					IssuerRef: cmmeta.ObjectReference{Name: "test"},
+				},
+			},
+			expectError: false,
+		},
+		"No errors on  valid certificaterequest with special usages set": {
+			input: &v1.CertificateRequest{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test",
+					Namespace: "default",
+				},
+				Spec: v1.CertificateRequestSpec{
+					Request: mustGenerateCSR(t, &v1.Certificate{
+						Spec: v1.CertificateSpec{
+							DNSNames: []string{"example.com"},
+							Usages:   []v1.KeyUsage{v1.UsageDigitalSignature, v1.UsageKeyEncipherment, v1.UsageClientAuth},
+						},
+					}),
+					Usages:    []v1.KeyUsage{v1.UsageDigitalSignature, v1.UsageKeyEncipherment, v1.UsageClientAuth},
+					IssuerRef: cmmeta.ObjectReference{Name: "test"},
+				},
+			},
+			expectError: false,
+		},
+		"Errors on certificaterequest with mismatch of usages": {
+			input: &v1.CertificateRequest{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test",
+					Namespace: "default",
+				},
+				Spec: v1.CertificateRequestSpec{
+					Request: mustGenerateCSR(t, &v1.Certificate{
+						Spec: v1.CertificateSpec{
+							DNSNames: []string{"example.com"},
+							Usages:   []v1.KeyUsage{v1.UsageDigitalSignature, v1.UsageKeyEncipherment, v1.UsageClientAuth},
+						},
+					}),
+					Usages:    []v1.KeyUsage{v1.UsageDigitalSignature, v1.UsageKeyEncipherment, v1.UsageCodeSigning},
+					IssuerRef: cmmeta.ObjectReference{Name: "test"},
+				},
+			},
+			expectError: true,
+			errorSuffix: "csr key usages do not match specified usages: [[2]: \"client auth\" != \"code signing\"]",
+		},
```

Comment:
Maybe add a test for cases where CSR has usages set but CR does not, and vice versa.